File: Java/약수의 합.java
// 약수의 합
public class Q25 {
    public static void main(String[] args) {
        int n = 12;
        DivSum divSum = new DivSum();
        System.out.println(divSum.divSum(n));
    }
}

class DivSum {
    public int divSum(int n) {
        int answer = 0;
        for(int i=1;i<=n;i++){
            if(n%i==0){
                answer+=i;
            }
        }
        return answer;
    }
}

File: JavaScript/약수의 합.js
/*
약수의 합 : 
  정수 n을 입력받아 n의 약수를 모두 더한 값을 리턴하는 함수, solution을 완성해주세요.
제한 사항 : 
  n은 0 이상 3000이하인 정수입니다.
*/

const solution = (n) =>
  Array.from(new Array(n), (x, i) => i + 1).reduce((a, c) => (n % c === 0 ? a + c : a + 0),0);

const sum = n => {
  let answer = 0;
  for (let i = 1; i <= n; i++) {
    if (n % i === 0) {
      answer += i;
    }
  }
  return answer;
};

console.log(sum(12)); // 28
console.log(solution(12)); // 28
